Sources/TuistKit/Commands/LintCommand.swift

```diff
@@ -0,0 +1,115 @@
+import Basic
+import Foundation
+import SPMUtility
+import TuistCore
+import TuistGenerator
+import TuistLoader
+import TuistSupport
+
+enum LintCommandError: FatalError, Equatable {
+    /// Thrown when neither a workspace or a project is found in the given path.
+    case manifestNotFound(AbsolutePath)
+
+    /// Error type.
+    var type: ErrorType {
+        switch self {
+        case .manifestNotFound:
+            return .abort
+        }
+    }
+
+    /// Description
+    var description: String {
+        switch self {
+        case let .manifestNotFound(path):
+            return "Couldn't find Project.swift nor Workspace.swift at \(path.pathString)"
+        }
+    }
+}
+
+/// Command that builds a target from the project in the current directory.
+class LintCommand: NSObject, Command {
+    /// Command name.
+    static var command: String = "lint"
+
+    /// Command description.
+    static var overview: String = "Lints a workspace or a project that check whether they are well configured."
+
+    /// Graph linter
+    private let graphLinter: GraphLinting
+    private let environmentLinter: EnvironmentLinting
+    private let manifestLoading: ManifestLoading
+    private let graphLoader: GraphLoading
+    let pathArgument: OptionArgument<String>
+
+    /// Default constructor.
+    public required convenience init(parser: ArgumentParser) {
+        let manifestLoader = ManifestLoader()
+        let generatorModelLoader = GeneratorModelLoader(manifestLoader: manifestLoader,
+                                                        manifestLinter: AnyManifestLinter())
```

Comment:
isn't it worth also linting manifests here?

Reply:
In fact, the first implementation used the manifest linter. However, if that validation fails it'd exit immediately and not run the other linters. Since the manifest linter just runs one validation, and I doubt we'll add more there. I think it's safe to leave it out. If we see we end up adding more validations there, we might consider changing the logic here.